test/tests_on_odes/l2_colloc_grad_test.jl

```diff
@@ -0,0 +1,20 @@
+using BlackBoxOptim
+
+
+cost_function = build_loss_objective(prob1,Tsit5(),L2Loss(t,data,colloc_grad=colloc_grad(t, data)),
+                                     maxiters=10000,verbose=false)
+bound1 = Tuple{Float64, Float64}[(1, 2)]
+result = bboptimize(cost_function;SearchRange = bound1, MaxSteps = 11e3)
+@test result.archive_output.best_candidate[1] ≈ 1.5 atol=3e-1
+
+cost_function = build_loss_objective(prob2,Tsit5(),L2Loss(t,data,colloc_grad=colloc_grad(t, data)),
+                                     maxiters=10000,verbose=false)
+bound2 = Tuple{Float64, Float64}[(1, 2),(1, 4)]
+result = bboptimize(cost_function;SearchRange = bound2, MaxSteps = 11e3)
+@test result.archive_output.best_candidate ≈ [1.5;3.0] atol=3e-1
+
+cost_function = build_loss_objective(prob3,Tsit5(),L2Loss(t,data,colloc_grad=colloc_grad(t, data)),
```

Comment:
I'd like to see one where the weight on the shooting part is set really small so that it's a unit test on the collocation part.

Reply:
The idea was that the collocation and weight parts don't intersect and if they work correctly separately they will work correctly together. Or do I miss something here?

Reply:
I don't think you've tested the separately?